### PR TITLE
Fixes Part Of #4194: Added dark mode support to DrawerFragment, ConceptCardFragment and AudioFragment

### LINKED ISSUE #4194: Add dark mode support to HomeActivity, DrawerFragment, RevisionCardActivity, ConceptCardFragment and AudioFragment
Add dark mode support to HomeActivity, DrawerFragment, RevisionCardActivity, ConceptCardFragment and AudioFragment

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by Akshatkamboj14**
## Explanation
Fixes Part Of #4194 

**Drawer Fragment :** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/eb119ee3-033f-4757-8aee-7ea234d6f126/specs/

**Concept Card Fragment :** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/2ee531cf-c358-4a7d-b9dd-b772b51c3d3c/specs/

**Audio Fragment :** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d67b268b-f6d9-434f-8897-24a2b13ccd84/specs/

**_these 2 needs to be done in following PRs._**

**Revision Card Activity :** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/5d907b23-a75e-4f8b-8c59-afd3c734c53e/specs/

**Home Activity :** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/67bfcbac-7b8e-491b-89c3-38937b37e5c2/specs/

- for future suggestion if somebody wants to show the Coming soon activity they can go to their project and can search for **_TopicListController.kt_** and comment out line number 266, 267 and 269. another suggestion is that they can complete all the chapters from developers options and complete recently played activities or they can refer to this document for further details -> https://docs.google.com/document/d/1o7hOz1rP-hJaKn1f5x6LV6ms0bzhyCpsTifJQY5crCc/edit#heading=h.ovlu0iazlug6

| **Default** | **Dark Mode** |
| ------------- | ------------- |
| image  | image  |
| image  | image  |
| image  | image  |
| image | image  |
| image  | image  |

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Akshatkamboj14** [comment]
@rt4914 I have a doubt about this, Idk how to add dark mode in the recently played stories text view and card. and meanwhile, in the mocks there are Coming soon Cards and text view, I have added dark mode support to that as well but IDK how to present that on the Oppia application, PTAL.
| Default  | Dark |
| ------------- | ------------- |
| image | image  |

File: app/src/main/res/drawable-ldrtl/bottom_left_rounded_rect_coming_soon_background.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     android:topRightRadius="0dp" />
   <gradient
     android:angle="360"
-    android:endColor="@color/coming_soon_end"
-    android:startColor="@color/coming_soon_start"
+    android:endColor="@color/color_def_grape_violet"
```
Comment (rt4914) [inline app/src/main/res/drawable-ldrtl/bottom_left_rounded_rect_coming_soon_background.xml:10]: Do not use color from `color_defs.xml` directly here. We should only use colors from `component_colors.xml` file.

File: app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     android:topRightRadius="4dp" />
   <gradient
     android:angle="360"
-    android:endColor="@color/coming_soon_end"
-    android:startColor="@color/coming_soon_start"
+    android:endColor="@color/color_def_grape_violet"
+    android:startColor="@color/color_def_bright_violet"
```
Comment (rt4914) [inline app/src/main/res/drawable/bottom_left_rounded_rect_coming_soon_background.xml:11]: Ditto

File: app/src/main/res/drawable/state_button_primary_background.xml [hunk 3]
```diff
@@ -6,4 +6,4 @@
   <stroke
     android:width="1dp"
     android:color="@color/component_color_continue_button_border_color" />
-</shape>
+</shape>
```
Comment (rt4914) [inline app/src/main/res/drawable/state_button_primary_background.xml:9]: EOF is missing, add one blank line at the end.

File: app/src/main/res/layout-sw600dp/revision_card_fragment.xml [hunk 4]
```diff
@@ -12,7 +12,7 @@
   <ScrollView
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/color_def_oppia_light_green"
+    android:background="@color/component_color_revision_card_activity_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp/revision_card_fragment.xml]: Revert changes from this file.

File: app/src/main/res/layout/audio_fragment.xml [hunk 5]
```diff
@@ -35,6 +35,7 @@
       android:contentDescription="@{viewModel.playStatusLiveData == UiAudioPlayStatus.PLAYING ? @string/audio_pause_description : @string/audio_play_description}"
       android:onClick="@{(v) -> viewModel.togglePlayPause(viewModel.playStatusLiveData)}"
       android:padding="12dp"
+      app:tint="@color/component_color_audio_fragment_icon_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/audio_fragment.xml:38]: This is incorrect as it is not a `background_color`. Please create a new value which makes sense here.

File: app/src/main/res/layout/audio_fragment.xml [hunk 6]
```diff
@@ -70,6 +71,7 @@
       android:onClick="@{(v) -> audioFragment.languageSelectionClicked()}"
       android:padding="12dp"
       app:srcCompat="@drawable/ic_audio_lang_24px"
+      app:tint="@color/component_color_audio_fragment_icon_background_color"
```
Comment (rt4914) [inline app/src/main/res/layout/audio_fragment.xml:74]: Ditto

File: app/src/main/res/layout/drawer_fragment.xml [hunk 7]
```diff
@@ -66,7 +68,7 @@
             android:layout_height="24dp"
             android:layout_gravity="center_vertical"
             app:srcCompat="@drawable/ic_baseline_code_24"
-            android:tint="@{footerViewModel.isDeveloperOptionsSelected ? @color/highlighted_developer_options_nav_menu_item : @color/oppia_primary_text_dark}"/>
+            android:tint="@{footerViewModel.isDeveloperOptionsSelected ? @color/color_def_persian_blue : @color/component_color_drawer_fragment_developer_options_text_color}"/>
```
Comment (rt4914) [inline app/src/main/res/layout/drawer_fragment.xml:71]: Do not use color from `color_defs.xml` file. Use colors only from `component_colors.xml` file

File: app/src/main/res/layout/drawer_fragment.xml [hunk 8]
```diff
@@ -75,7 +77,7 @@
             android:layout_marginStart="12dp"
             android:fontFamily="sans-serif-medium"
             android:text="@string/developer_options"
-            android:textColor="@{footerViewModel.isDeveloperOptionsSelected ? @color/highlighted_developer_options_nav_menu_item : @color/oppia_primary_text_dark}"
+            android:textColor="@{footerViewModel.isDeveloperOptionsSelected ? @color/color_def_persian_blue : @color/component_color_drawer_fragment_developer_options_text_color}"
```
Comment (rt4914) [inline app/src/main/res/layout/drawer_fragment.xml:80]: Ditto

File: app/src/main/res/layout/drawer_fragment.xml [hunk 9]
```diff
@@ -97,7 +99,7 @@
             android:layout_height="24dp"
             android:layout_gravity="center_vertical"
             app:srcCompat="@drawable/ic_admin_settings_icon_brown_24dp"
-            android:tint="@{footerViewModel.isAdministratorControlsSelected ? @color/highlighted_nav_menu_item : @color/oppia_primary_text_dark}" />
+            android:tint="@{footerViewModel.isAdministratorControlsSelected ? @color/color_def_oppia_brown_dark : @color/component_color_drawer_fragment_admin_controls_text_color}" />
```
Comment (rt4914) [inline app/src/main/res/layout/drawer_fragment.xml:102]: Ditto

File: app/src/main/res/layout/drawer_fragment.xml [hunk 10]
```diff
@@ -106,7 +108,7 @@
             android:layout_marginStart="12dp"
             android:fontFamily="sans-serif-medium"
             android:text="@string/administrator_controls"
-            android:textColor="@{footerViewModel.isAdministratorControlsSelected ? @color/highlighted_nav_menu_item : @color/oppia_primary_text_dark}"
+            android:textColor="@{footerViewModel.isAdministratorControlsSelected ? @color/color_def_oppia_brown_dark : @color/component_color_drawer_fragment_admin_controls_text_color}"
```
Comment (rt4914) [inline app/src/main/res/layout/drawer_fragment.xml:111]: Ditto

File: app/src/main/res/values-v21/styles.xml [hunk 11]
```diff
@@ -9,9 +9,9 @@
   </style>
 
   <style name="AudioSeekBar" parent="Widget.AppCompat.SeekBar">
-    <item name="android:progressBackgroundTint">@color/progress_background_tint</item>
-    <item name="android:progressTint">@color/color_def_white</item>
-    <item name="android:colorControlActivated">@color/progress_background_tint</item>
+    <item name="android:progressBackgroundTint">@color/component_color_audio_fragment_seekbar_progress_background_color</item>
+    <item name="android:progressTint">@color/component_color_audio_fragment_icon_background_color</item>
```
Comment (rt4914) [inline app/src/main/res/values-v21/styles.xml:13]: This is not a `background_color`

File: app/src/main/res/values-v21/styles.xml [hunk 12]
```diff
@@ -9,9 +9,9 @@
...
+    <item name="android:colorControlActivated">@color/component_color_audio_fragment_seekbar_progress_background_color</item>
```
Comment (rt4914) [inline app/src/main/res/values-v21/styles.xml:14]: This is not a `background_color`

File: app/src/main/res/values/component_colors.xml [hunk 13]
```diff
@@ -96,4 +96,19 @@
   <!-- License Text Viewer Activity -->
   <color name="component_color_license_text_viewer_activity_background_color">@color/color_palette_dark_background_color</color>
   <color name="component_color_license_text_viewer_activity_text_color">@color/color_palette_description_text_color</color>
+  <!-- Audio Fragment-->
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml:156]: Space missing
`<!-- Audio Fragment -->`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Akshatkamboj14 Suggested changes.

Also, please remove `HomeActivity` from PR title
and fix the color of Concept-card status bar too.

File: app/src/main/res/drawable-ldrtl/bottom_left_rounded_rect_coming_soon_background.xml (hunk 1 above)
Comment (Akshatkamboj14) [reply app/src/main/res/drawable-ldrtl/bottom_left_rounded_rect_coming_soon_background.xml:10]: @rt4914 like in this only one color is used and its also not changed in the dark mode so can I use this color or make a new value in the component colors?

Comment (rt4914) [reply app/src/main/res/drawable-ldrtl/bottom_left_rounded_rect_coming_soon_background.xml:10]: You will need to create new value in components for such case too.

**Pull request closed without merging**